.github/workflows/ci.yaml

```diff
@@ -77,3 +77,29 @@ jobs:
 
       - name: Check for unstaged files
         run: git diff --exit-code
+  fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Echo Go Cache Paths
+        id: go-cache-paths
+        run: |
+          echo "GOCACHE=$(go env GOCACHE)" >> ${{ runner.os == 'Windows' && '$env:' || '$' }}GITHUB_OUTPUT
+          echo "GOMODCACHE=$(go env GOMODCACHE)" >> ${{ runner.os == 'Windows' && '$env:' || '$' }}GITHUB_OUTPUT
+
+      - name: Go Build Cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.go-cache-paths.outputs.GOCACHE }}
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.**', '**.go') }}
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "~1.21"
+
+      - name: Check format
+        run: test -z $(go run mvdan.cc/gofumpt -l .)
```

Comment:
Do you have an example of a failed CI run?

Reply:
Since this is just a boolean check, perhaps it can be more helpful if we show what went wrong? How about:

```shell
list=$(go run mvdan.cc/gofumpt -l .)
if [[ -n $list ]]; then
	echo -e "error: The following files have changes:\n\n${list}\n\nDiff:\n\n"
	go run mvdan.cc/gofumpt -d .
	exit 1
fi
```

Reply:
Using @mafredri suggestion the output would look like this:
```
error: The following files have changes:

envbuilder.go

Diff:


diff envbuilder.go.orig envbuilder.go
--- envbuilder.go.orig
+++ envbuilder.go
@@ -151,7 +151,7 @@
                if err != nil {
                        return fmt.Errorf("parse docker config: %w", err)
                }
-               err = os.WriteFile(filepath.Join(MagicDir, "config.json"), decoded, 0644)
+               err = os.WriteFile(filepath.Join(MagicDir, "config.json"), decoded, 0o644)
                if err != nil {
                        return fmt.Errorf("write docker config: %w", err)
                }
```

Reply:
Looks like I added one \n too much 😅, and maybe the "Diff:" is superfluous. Thanks for trying it out @BrunoQuaresma!